apollo-router/src/axum_factory/listeners.rs

```diff
@@ -234,16 +262,10 @@ pub(super) fn serve_router_on_listen_addr(
                                 tracing::info!("can accept connections again");
                                 MAX_FILE_HANDLES_WARN.store(false, Ordering::SeqCst);
                             }
+
                             // We only want to recognise sessions if we are the main graphql port.
-                            if main_graphql_port {
-                                let session_count = SESSION_COUNT.fetch_add(1, Ordering::Acquire)+1;
-                                tracing::info!(
-                                    value.apollo_router_session_count_total = session_count,
-                                    listener = &address
-                                );
-                            }
+                            let _guard = main_graphql_port.then(TotalSessionCountGuard::start);
```

Comment:
Note this contains a mistake. The guard must be moved into the `task::spawn` call below, but it isn't, so the total session count metric is broken with this PR (it is immediately decremented before the next incoming request is handled.)

https://github.com/apollographql/router/pull/6527 will fix this problem.

Reply:
That PR will be redone in a different way. https://github.com/apollographql/router/pull/6539 fixes just this problem.